[Feature/ pyokemon-34] 공연예매 UI/API 연동

seoyoun8694 opened the pull request:
## ✏️ 작업 개요  
공연 예매 UI/API 연동

## 🔨 작업 내용 상세
### 공연 예매 페이지 1: 전체 공연장 배치도
- **좌석 등급별 시각적 구분**
  - VIP, R, A, B 등급을 색상으로 구분
- **등급별 인터랙션**
  - 각 등급 영역 클릭 시 해당 좌석 등급의 상세 배치도로 이동
- **반응형 레이아웃**
  - CSS Grid 활용

### 공연 예매 페이지 2: 좌석 등급별 상세 좌석 배치도
- **좌석 렌더링**
  - API로부터 받은 데이터를 기반으로 좌석 구성 렌더링
- **좌석 상태 표시**
  - 예약 가능: 기본 스타일  
  - 선택됨: 테두리 강조  
  - 예약됨: 비활성화
- **좌석 선택 기능**
  - 클릭 시 선택/해제 가능
- **좌석 정보 표시**
  - 행/열 번호, 좌석 ID 등 표시

###  사이드바 컴포넌트
- **좌석 등급 정보**
  - 각 등급별 가격, 잔여석 수 표시
- **선택 좌석 정보**
  - 사용자가 선택한 좌석의 상세 정보 표시
- **결제 기능**
  - 선택 완료 후 결제 진행 버튼 제공

## 🔗 관련 이슈  
- Closes #5 

## ✅ 체크리스트  
- [x] 전체 공연장 배치도
- [x] 사이드바
    - [ ] 미니맵
    - [x] 좌석 등급/잔여석
    - [x] 선택 좌석
- [x] 해당 구역(seat-class)의 좌석 배치도
- [x] 특정 좌석 선택 예매
- [x] API 연결
    - [x] 전체 좌석 배치도 - 잔여 좌석 (/api/events/booking/{eventId})
    - [x] VIP 좌석 배치도 - 좌석 status (/api/events/booking/{eventId}/{seatGrade})

## 💬 기타 참고 사항 
- 추가 라이브러리 없음
- 접속 URL: http://localhost:5173/event/booking/1
- VIP만 선택 가능!! -> 데베에 공연장 정보 있어야함!! 없는 사람 슬랙으로 연락바람~
- 파일 구조
```text
src/
├── pages/event-booking/
│   ├── booking-page.tsx          # 메인 예매 페이지
│   └── _component/
│       ├── seating_chart.tsx     # 전체 공연장 배치도
│       ├── seat_class_seat.tsx   # 구역별 상세 좌석 배치도
│       └── booking_sidebar.tsx   # 사이드바
├── api/booking/
│   ├── fetchers/get-event-booking.ts
│   └── queries/use-get-event-booking-query.ts
└── types/booking.ts              # 타입 정의
```

seoyoun8694 committed:
[PYOKEMON-34] [feature] 목업 데이터 설정 #5

seoyoun8694 committed:
[PYOKEMON-34] [feature] 전체 좌석 배치도 UI #5

seoyoun8694 committed:
[PYOKEMON-34] [feature] VIP 좌석 배치도 UI #5

seoyoun8694 committed:
[PYOKEMON-34] [feature] sidebar UI #5

seoyoun8694 committed:
[PYOKEMON-34] [feature] 공연 예매 페이지 routes 연결 #5

seoyoun8694 committed:
[PYOKEMON-34] [Feature] 전체 잔여 좌석 api 연결 #5

seoyoun8694 committed:
[PYOKEMON-34] [Feature] VIP 좌석 status api 연결 #5

- useEffect 대신 useCallback 사용

seoyoun8694 committed:
[PYOKEMON-34] [fix] api 연결 수정 #5

- query에서 data 받아오기
- useMemo 사용

seoyoun8694 committed:
[PYOKEMON-34] [fix] porps 변경 #5

sky121016 commented on src/api/booking/fetchers/get-event-booking.ts:
```diff
@@ -0,0 +1,25 @@
+  }
+}
+
+export const getSeat_class = async (eventScheduleId: number, seatGrade: string) => {
```
getSeat_class
Camel case로 바꾸는 거 어떨까요??

soobin70 reviewed (commented):
booking.ts 삭제해주세요!

sky121016 commented on src/api/booking/queries/use-get-event-booking-query.ts:
```diff
@@ -0,0 +1,17 @@
+
+export const useGetSeatClassQuery = (eventScheduleId: number, seatGrade: string) => {
+  return useQuery({
+    queryKey: ["seatClass", eventScheduleId.toString(), seatGrade],
```
toString() 없애도 될 것 같아요

seoyoun8694 committed:
[PYOKEMON-34] [fix] 코드리뷰 수정 #5

- booking constants 수정
- queries - toString 변경
- getSeat_class - Camel case로 변경(getSeatClass)

dldh1218 reviewed (approved):
React Query hook 상태 관리랑 Booking page 컴포넌트 분리가 잘 되어있는 거 같습니다!
고생하셨어요~!! 👍👍👍

seoyoun8694 committed:
Merge branch 'develop' into feature/-PYOKEMON-34-공연예매
